---
title: "Final_DEG"
author: "Huyu Liu"
date: "12/9/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


#Loading packages
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
#BiocManager::install("DESeq")
#BiocManager::install("tximport")
#install.packages("ggplot2")
#install.packages("reshape2")
#install.packages("RColorBrewer")
library(DESeq)
library(tximport)
library("ggplot2")
library(reshape2)
library(RColorBrewer)
```

#Convert results from RSEM

```{r}
getwd()
setwd("/Users/huyusmac/Documents/class/BCB546X/BCB546X_FinalGroup/RNAseq_heatmap")
samples <- read.csv("samples.csv")
files <- file.path("/Users/huyusmac/Documents/class/BCB546X/BCB546X_FinalGroup/individual", paste0(samples$X, ".genes.results"))
names(files) <- samples$X
txi.rsem <- tximport(files, type = "rsem", txIn = FALSE, txOut = FALSE)
#head(txi.rsem$counts)
```


#Expression level

```{r}
CountTable <- txi.rsem$counts
for(a in c(1:73)){
  for(b in c(1:33)){
    CountTable[a,b]<-as.integer(CountTable[a,b])
  }
} #convert to integer

# change gene id into gene name
genelist <- read.csv("genes.csv")
rownames(genelist)<-genelist[,1]
genelist<-genelist
CountTable <- cbind(CountTable,genelist)
rownames(CountTable)<-CountTable$gene
CountTable <- CountTable[,c(1:33)]

#get expression level
stage = samples$stage
cds<-newCountDataSet(CountTable, stage)
cds <-estimateSizeFactors( cds ) #count size factor
cds = estimateDispersions(cds) 

count<-counts(cds, normalized=TRUE)
```


#Differential expressed gene

```{r}
# statistical significance of gene expression in apical region
resm = nbinomTest(cds, "nonreproductive top", "nonreproductive middle")
resb = nbinomTest( cds, "nonreproductive top", "nonreproductive bottom" )

write.csv(resm, file="DEG top_middle.csv" )
write.csv(resm, file="DEG top_bottom.csv" )
```

#Heatmap

```{r}
#average replications
count<-as.data.frame(count)
count$non_rep_bot<-(count$non_rep_bot_1+count$non_rep_bot_2+count$non_rep_bot_3)/3
count$non_rep_mid<-(count$non_rep_mid_1+count$non_rep_mid_2+count$non_rep_mid_4+count$non_rep_mid_5+count$non_rep_mid_6)/5
count$non_rep_top<-(count$non_rep_top_1+count$non_rep_top_2+count$non_rep_top_3)/3
count$early_cleavage<-(count$early_cleavage_1+count$early_cleavage_2)/2
count$early_postinversion<-(count$early_postinversion_1+count$early_postinversion_2)/2
count$early_preinversion<-(count$early_preinversion_1+count$early_preinversion_2)/2
count$early_vitellogenesis<-(count$early_vitellogenesis_1+count$early_vitellogenesis_2+count$early_vitellogenesis_3)/3
count$fertilization<-(count$fertilization_1+count$fertilization_2)/2
count$late_cleavage<-(count$late_cleavage_1+count$late_cleavage_2)/2
count$late_postinversion<-count$late_postinversion_1
count$late_preinversion<-(count$late_preinversion_1+count$late_preinversion_2)/2
count$late_vitellogenesis<-(count$late_vitellogenesis_1+count$late_vitellogenesis_2+count$late_vitellogenesis_3+count$late_vitellogenesis_4)/4
count$swimming_larvae<-(count$swimming_larvae_1+count$swimming_larvae_2)/2
count<-count[,c(34:46)]
colnames(count)<-c("nonreproductive bottom", "nonreproductive middle", "nonreproductive top", "early cleavage", "early postinversion", "early preinversion", "early vitellogenesis", "ferilization", "late cleavage" , "late postinversion", "late preinversion", "late vitellogenesis", "swimming larvae")

#convert data format
count$gene<-rownames(count)
count.m <- melt(count)
colnames(count.m)<-c("gene","stage","expression")

#set color breaks
summary_v <- summary(count.m$expression)
break_v <- unique(c(seq(summary_v[1]-0.05,summary_v[2],length=4),seq(summary_v[2],summary_v[3],length=3),seq(summary_v[3],summary_v[5],length=3),seq(summary_v[5],summary_v[6]*1.05,length=4)))
break_v<-round(as.numeric( as.character(break_v)),2)
count.m$expression <- cut(count.m$expression, breaks=break_v,labels=break_v[2:length(break_v)])
break_v<-unique(count.m$expression)
gradientC=c('royalblue','yellow','red')
col <- colorRampPalette(gradientC)(length(break_v))

#make and save heatmap
p<- ggplot(count.m,aes(stage,gene,fill= expression)) + geom_tile(colour = "white") +scale_x_discrete(position = "top") + theme(axis.text.x = element_text(size= 12, angle = 90, hjust = 0, ), axis.text.y = element_text(hjust=1))
p <- p + scale_fill_manual(values=col)+ xlab("") + ylab("")
ggsave("heatmap.jpg",width = 20, height = 35, units ="cm", dpi = 300)
```

